flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkStateSnapshotUtils.java

```diff
@@ -0,0 +1,382 @@
+                        .jobReference(JobReference.fromFlinkResource(resource))
+                        .savepoint(savepointSpec)
+                        .build();
+
+        var resourceName = getFlinkStateSnapshotName(SAVEPOINT, triggerType, resource);
+        return createFlinkStateSnapshot(kubernetesClient, resourceName, snapshotSpec, triggerType);
+    }
+
+    /**
+     * Creates a checkpoint {@link FlinkStateSnapshot} resource on the Kubernetes cluster.
+     *
+     * @param kubernetesClient kubernetes client
+     * @param resource Flink resource associated
+     * @param checkpointType type of checkpoint
+     * @param triggerType trigger type
+     * @return created snapshot
+     */
+    public static FlinkStateSnapshot createCheckpointResource(
+            KubernetesClient kubernetesClient,
+            AbstractFlinkResource<?, ?> resource,
+            CheckpointType checkpointType,
+            SnapshotTriggerType triggerType) {
+        var checkpointSpec = CheckpointSpec.builder().checkpointType(checkpointType).build();
+
+        var snapshotSpec =
+                FlinkStateSnapshotSpec.builder()
+                        .jobReference(JobReference.fromFlinkResource(resource))
+                        .checkpoint(checkpointSpec)
+                        .build();
+
+        var resourceName = getFlinkStateSnapshotName(CHECKPOINT, triggerType, resource);
+        return createFlinkStateSnapshot(kubernetesClient, resourceName, snapshotSpec, triggerType);
+    }
+
+    /**
+     * Based on job configuration and operator configuration, decide if {@link FlinkStateSnapshot}
+     * resources should be used or not. Operator configuration will disable the usage of the
+     * corresponding CRD was not installed on this Kubernetes cluster.
+     *
+     * @param operatorConfiguration operator config
+     * @param configuration job config
+     * @return true if snapshot resources should be created
+     */
+    public static boolean isSnapshotResourceEnabled(
+            FlinkOperatorConfiguration operatorConfiguration, Configuration configuration) {
+        return configuration.get(SNAPSHOT_RESOURCE_ENABLED)
+                && operatorConfiguration.isSnapshotResourcesEnabled();
+    }
+
+    /**
+     * Return a generated name for a {@link FlinkStateSnapshot} to be created.
+     *
+     * @param snapshotType type of snapshot
+     * @param triggerType trigger type of snapshot
+     * @param referencedResource referenced resource
+     * @return result name
+     */
+    public static String getFlinkStateSnapshotName(
+            SnapshotType snapshotType,
+            SnapshotTriggerType triggerType,
+            AbstractFlinkResource<?, ?> referencedResource) {
+        return String.format(
+                "%s-%s-%s-%s",
+                referencedResource.getMetadata().getName(),
+                snapshotType.name().toLowerCase(),
+                triggerType.name().toLowerCase(),
+                UUID.randomUUID());
```

Comment:
Hi @mateczagany, wdyt about having a timestamp / epoch value instead of the UUID here? It might help users figure out the latest snapshot CR in case they do a
`kubectl get flinkstatesnapshots.flink.apache.org`  (and not have to pass in `--sort-by=.metadata.creationTimestamp` explicitly) ?

Reply:
Hi @anupamaggarwal you are right, that makes more sense than a random UUID. I have pushed a commit for this.